src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,67 @@
+package roomescape.service;
+
+import java.time.Clock;
+import java.time.LocalDate;
+import java.util.List;
+import org.springframework.stereotype.Service;
+import roomescape.dao.ReservationDao;
+import roomescape.dao.ReservationTimeDao;
+import roomescape.dao.ThemeDao;
+import roomescape.dto.ReservationRequest;
+import roomescape.dto.ReservationResponse;
+import roomescape.model.Reservation;
+import roomescape.model.ReservationTime;
+import roomescape.model.Theme;
+
+@Service
+public class ReservationService {
+
+    private final ReservationDao reservationDao;
+    private final ReservationTimeDao reservationTimeDao;
+    private final ThemeDao themeDao;
+    private final Clock clock;
+
+    public ReservationService(ReservationDao reservationDao,
+                              ReservationTimeDao reservationTimeDao,
+                              ThemeDao themeDao,
+                              Clock clock) {
+        this.reservationDao = reservationDao;
+        this.reservationTimeDao = reservationTimeDao;
+        this.themeDao = themeDao;
+        this.clock = clock;
+    }
+
+    public ReservationResponse addReservation(ReservationRequest reservationRequest) {
+        ReservationTime reservationTime = reservationTimeDao.findById(reservationRequest.timeId())
+                .orElseThrow(() -> new IllegalStateException("존재하지 않는 시간입니다."));
+        Theme theme = themeDao.findById(reservationRequest.themeId())
+                .orElseThrow(() -> new IllegalStateException("존재하지 않는 테마입니다."));
+        Reservation reservation = reservationRequest.toEntityWithReservationTime(reservationTime, theme);
+        if (reservation.isPast(LocalDate.now(clock))) {
```

Comment:
```suggestion
    public ReservationResponse addReservation(ReservationRequest reservationRequest) {
        Reservation reservation = reservationOf(reservationRequest);
        if (reservation.isPast(LocalDate.now(clock))) {
```

메서드 분리를 통해, 서비스의 주요메서드가 읽기 편하도록 해보면 어떨까요?